include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -179,6 +189,156 @@ void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
   }
 }
 
+template <class T>
+void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    comm::CommunicatorGrid grid, Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus) {
+  using hpx::unwrapping;
+
+  auto executor_hp = dlaf::getHpExecutor<Backend::MC>();
+  auto executor_np = dlaf::getNpExecutor<Backend::MC>();
+
+  // Set up MPI
+  auto executor_mpi = dlaf::getMPIExecutor<Backend::MC>();
+  common::Pipeline<comm::Communicator> mpi_col_task_chain(grid.colCommunicator());
+  common::Pipeline<comm::Communicator> mpi_row_task_chain(grid.rowCommunicator());
+
+  const SizeType c_local_rows = mat_c.distribution().localNrTiles().rows();
+  const SizeType c_local_cols = mat_c.distribution().localNrTiles().cols();
+  const SizeType v_local_rows = mat_v.distribution().localNrTiles().rows();
+  const SizeType m = mat_c.nrTiles().rows();
+  const SizeType n = mat_c.nrTiles().cols();
+  const SizeType mb = mat_v.blockSize().rows();
+
+  auto dist_v = mat_v.distribution();
+  auto dist_c = mat_c.distribution();
```

Comment:
If you move `dist_v`/`dist_c` above, you can use them also for the rest of the initialization.

IMHO `c_local_rows`, `c_local_cols` and `v_local_rows` (the latter one is equal to the former one by constraint) do not improve the readability, I would say they make it even harder to understand (`_local_rows` is number of elements or number of tiles?).

Whatever the decision will be, I would try to use the same style: in certain cases these variables are used, in others `localNrTiles()` is used.

Reply:
So far I've just tried to use the same style in the whole code. What should we do? Keep the `local_rows`/`local_cols` or is it more readable if we use explicitly `localNrTiles()`?

Reply:
I don't think that `local_rows` and `local_cols` can be considered as part of the style.
It is used only in one place https://github.com/eth-cscs/DLA-Future/blob/1155b0c7202a6122ea59589cddeb026b85b0968b/include/dlaf/solver/triangular/impl.h#L346-L348
and looking again at these lines I would not consider the meaning given to them readable...